server/performance.py

```diff
@@ -0,0 +1,92 @@
+from typing import TYPE_CHECKING
+from typing import TypedDict
+
+from akatsuki_pp_py import Beatmap as CalculatorBeatmap
+from akatsuki_pp_py import Calculator
+
+if TYPE_CHECKING:
+    from server.repositories.beatmaps import Beatmap
+    from server.repositories.scores import Score
+
+
+class DifficultyAttributes(TypedDict):
+    star_rating: float
+    aim_star_rating: float | None
+    speed_star_rating: float | None
+    flashlight_star_rating: float | None
+    slider_star_rating: float | None
+    speed_note_count: float | None
+    ar: float | None
+    od: float | None
+    stamina_star_rating: float | None
+    color_star_rating: float | None  # TODO: what is this? james will know
```

Comment:
each mode has their own set of skills that are used to calculate difficulty. in taiko, one of the skills is called `color`. that's what this is. fwiw, i would split these into multiple models for each mode as they are all defined separately..

Reply:
right, interesting.. then i suppose things don't have to all be optional

know where i can find which attributes are a part of which modes?

Reply:
i see, i can read into https://github.com/osuAkatsuki/akatsuki-pp-rs